spec/unit/middleware/service_broker_rate_limiter_spec.rb

```diff
@@ -115,6 +109,80 @@ module Middleware
             end
           end
         end
+
+        shared_examples 'rate-limited endpoint' do
+          it 'does not allow more than the max number of concurrent requests' do
+            allow(app).to receive(:call) do
+              sleep(1)
+              [200, {}, 'a body']
+            end
```

Comment:
I'd move that into a before block.
Would be nice if we could get rid of the sleep but could be difficult